### PR TITLE
Add playground for interpolateColor function

### PR DESCRIPTION
**Pull request by tboba**

### CODE REVIEW

File: docs/animations/withTiming.mdx [hunk 1]
```diff
@@ -53,6 +53,7 @@ The value on which the animation will come at rest.
 The timing animation configuration.
 
 import { useTimingPlayground } from "@site/src/components/ModifierPlayground";
+import { useInterpolateColorPlayground } from "@site/src/components/ModifierPlayground";
```
Comment (kacperkapusciak) [inline docs/animations/withTiming.mdx:56]: You can add a new `interpolateColor.mdx` page to Advanced APIs section. If you don't want it to be shown after merge, but the code to be on main, you can prefix it with `_` like so: `_interpolateColor.mdx`.

That way you can iterate on the feature and merge it to main without it being seen by the end user.

File: src/components/ModifierPlayground/useInterpolateColorPlayground/ColorPicker/index.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+          const value = e.target.value;
+          if (!value) {
+            setInputColor("FFF");
+            setColor("#FFF");
+          }
+        }}
+        onChange={(e) => {
+          const value = e.target.value;
+          const hexPattern = /^([0-9A-F]{3}){1,2}$/i;
+          const enteringHexPattern = /^[0-9A-F]{0,6}$/;
```
Comment (kacperkapusciak) [inline src/components/ModifierPlayground/useInterpolateColorPlayground/ColorPicker/index.tsx:55]: User should be able to input lowercase letters too

File: src/components/ModifierPlayground/useInterpolateColorPlayground/Example.tsx
Comment (kacperkapusciak) [inline src/components/ModifierPlayground/useInterpolateColorPlayground/Example.tsx:1]: I've tried to compare the new playground with the old one and it seems like, sadly, it doesn't work the same :( 

https://github.com/software-mansion-labs/reanimated-beta-docs/assets/39658211/18adfdcf-a2de-4e21-b690-a83eb5094781




https://github.com/software-mansion-labs/reanimated-beta-docs/assets/39658211/56cdd3b3-1166-4d58-9eae-44ab60c821bf

The table of colors is useful here. You can see that on the original one there's a "flick" of purple between red and blue in HSV. I can't see this in the new playground.

File: package.json [hunk 3]
```diff
@@ -29,13 +29,15 @@
     "@emotion/styled": "^11.10.6",
     "@mdx-js/react": "^1.6.22",
     "@mui/material": "^5.12.0",
+    "@types/react-color": "^3.0.6",
```
[the remark this answers is not in the document]
Comment (tboba) [reply package.json:32]: Yeah, I've been testing react-color before using react-colorful 😄 currently this change waits locally for push!

File: src/components/ModifierPlayground/useInterpolateColorPlayground/ColorPicker/index.tsx (hunk 2 above)
Comment (tboba) [reply src/components/ModifierPlayground/useInterpolateColorPlayground/ColorPicker/index.tsx:55]: I thought about it, I've added `i` at the end of pattern 👍

File: src/components/ModifierPlayground/useInterpolateColorPlayground/Example.tsx
Comment (tboba) [reply src/components/ModifierPlayground/useInterpolateColorPlayground/Example.tsx:1]: Yeah, looks like `ColorSpace` was not properly read before 🙈

File: docs/animations/withTiming.mdx (hunk 1 above)
Comment (tboba) [reply docs/animations/withTiming.mdx:56]: Sure, should I document it, or you will take the lead?

File: src/components/ModifierPlayground/index.tsx [hunk 4]
```diff
@@ -20,7 +21,11 @@ import {
...
 } from "@mui/material";
 
-export { useSpringPlayground, useTimingPlayground };
+export {
+  useSpringPlayground,
+  useTimingPlayground,
+  useInterpolateColorPlayground,
+};
 
 export default function ModifierPlayground(props: any) {
```
[the remark this answers is not in the document]
Comment (tboba) [reply src/components/ModifierPlayground/index.tsx:30]: I also haven't seen that 😅 I will!

File: docs/animations/withTiming.mdx (hunk 1 above)
Comment (tboba) [reply docs/animations/withTiming.mdx]: Okay, let's maybe merge this PR and create a new branch where we could document this function

**Pull request merged**